src/track/globaltrackcache.cpp

```diff
@@ -695,6 +754,11 @@ TrackRef GlobalTrackCache::initTrackId(
     EvictAndSaveFunctor* pDel = std::get_deleter<EvictAndSaveFunctor>(strongPtr);
     DEBUG_ASSERT(pDel);
 
+    m_mutex.lock();
```

Comment:
Where is mutex being unlocked? Is this done in a nested call?

Reply:
> Where is mutex being unlocked? Is this done in a nested call?

The `m_mutex.lock()` should be removed here, and moved to [L206](https://github.com/mixxxdj/mixxx/blob/0f633489f77f8624c96ac64c88b8ad0ff8f8017f/src/track/globaltrackcache.cpp#L206) (`GlobalTrackCacheResolver::initTrackIdAndUnlockCache()`) instead.